### ISSUE TITLE
Windows Client invokes unexpected reboot.

### ISSUE THREAD

**Issue by 2E0PGS** [keybase/client#3875]
Keybase GUI Version: 1.0.16-20160713110836+a1c00b3

Here is a copy of the Windows Logs / System:

The process C:\Users\MYUSER\AppData\Local\Temp\{F36C4DE3-13A8-4897-B12D-59749802E2C9}\.cr\DokanSetup_redist.exe (MYPCDOMAINNAME) has initiated the restart of computer MYPCDOMAINNAME on behalf of user MYPCDOMAINNAME\MYUSER for the following reason: Application: Installation (Planned)
 Reason Code: 0x80040002
 Shut-down Type: restart
 Comment: 

This caused my computer to random reboot. Maybe its due to the fact that I was typing while the program installed and the popup dialog that asked if its ok to reboot must have gotten activated? I presume there was meant to be a popup. However I suggest you don't "focus" the "control" that is the "yes" button. Other than that don't focus the form is another option.

**Comment by taruti** [comment]
@zanderz this is for you?

**Comment by zanderz** [comment]
We have a known issue with these reboots caused by the Visual C++ Runtime Redistributable package which has been addressed in an upcoming update. On the plus side, this particular reboot won't happen to you again until our dependency changes to another runtime.  Sorry about that!

**Comment by 2E0PGS** [comment]
Yes I was not sure if it was due to focused controls or the program simply just reboots no matter what. I did not have time to fire up a VM and test the installer again to check. However you have confirmed my original guess. Ok well its good you know about the issue. It's not good since I nearly lost some data. Luckily I had a auto save. Thanks for taking note of the issue. I think this should stay open until its resolved properly.

**Comment by jmerriweather** [comment]
I also had a forced reboot due to this issue and almost lost some work.

So you've added `/norestart` to the runtime redist installation in an upcoming update so it wont force restart?

**Comment by zanderz** [comment]
The approach is to not run the runtime redist package silently, so that its reboot prompt gets shown to the user.  Available in an upcoming release, such as https://prerelease.keybase.io/windows/Keybase_1.0.17-20160820134520%2B9aefaa4.386.exe if you'd like to try it now.

**Comment by osvaldsson** [comment]
I'm having the exact same problem and it has resulted in me uninstalling the keybase client.

Yesterday I installed the latest version but did not want to restart yet.

It should be my decision when to restart and it should never be forced by the client. If a forced restart is required then this should be made clear when the install client is run.

**Comment by maxtaco** [comment]
@osvaldsson thanks for your feedback, are you saying that you did get an unexpected reboot with the most recent version (linked directly above?)

**Comment by osvaldsson** [comment]
@maxtaco nope, I'm using the latest released version, trying to avoid the reboot still.

I'll do a reboot now, install that version and see how it goes as I have some minutes to spare.

**Comment by osvaldsson** [comment]
Probably totally unrelated but I had 8 separate install dialogs on my desktop when logged in this morning even after installing the update yesterday.

**Comment by osvaldsson** [comment]
@maxtaco I'll have to correct myself as when I tried to install the prerelease version listed above I get an error:

`Error 0x80070666: Cannot install a product when a newer version is installed.`

**Comment by maxtaco** [comment]
cc: @zanderz

**Comment by zanderz** [comment]
If someone has experienced a sudden reboot with no dialog box requesting it, I would be interested to see the installer bundle logs from the time in question, located at `%temp%\Keybase*.log`. We've seen that the visual studio runtime redistributable is the culprit, so we made it not invisible, and if it's something else we need to know.

**Comment by joshpj** [comment]
I just got an unprompted reboot after accepting an update.  "keybase version" says I'm now on 1.0.17-20160822152935+692af12, so I presume this should have been after the fix.

I've attached the %TEMP%\Keybase*.log files (and keybaseInstallLayout*) that I had with a modified date of today: [keybase_update_logs_joshpj_2016-08-25.zip](https://github.com/keybase/client/files/438358/keybase_update_logs_joshpj_2016-08-25.zip)

**Comment by osvaldsson** [comment]
I'm sorry but there is a high number of logs in that directory and none seem to point out anything obvious to me but I can't upload them here without reading through them all to check for sensitive data.

I'll remove those that are there now and will upload if it happens again.

**Comment by zanderz** [comment]
Since this thread was started, another unprompted reboot cause has been identified: When an update arrives which requires reboot, if you leave the "automatically update" checkbox checked, begin the update, then delay rebooting for at least an hour, an unprompted reboot may occur when the updater process does its check. A forthcoming update will address this, then next time drivers are updated, there should be no more unprompted reboots.

**Comment by osvaldsson** [comment]
I'm no longer experiencing random restart requests, now I just get a random "Modify Setup" window as stated in #4186

**Comment by dyfrgi** [comment]
I experienced this today, so it seems that this bug is still extant. Didn't see anything saying it required a reboot. Clicked "update" (leaving automatically update checked), went about my business, sudden reboot without a prompt about 3 minutes later.

[keybase_logs_20170219_dyfrgi.zip](https://github.com/keybase/client/files/786007/keybase_logs_20170219_dyfrgi.zip)

Should I expect this to happen on automatic updates as well?

**Comment by zanderz** [comment]
No, this should never happen, sorry and thanks for the report.

**Comment by zanderz** [comment]
https://keybase.atlassian.net/browse/CORE-4630

**Comment by cinderblock** [comment]
This may have happened to me today.

Today I was prompted to update Keybase client. I clicked ok. When it finished, it wanted to reboot. I said later because I had some work open. Hours later, I left the computer for ~30 min and came back to a rebooted machine and having lost some work. 😢 

This may have been Windows 10 installing an update. However, Windows has been pretty good about warning about impending reboots recently.

**Comment by zanderz** [comment]
@cinderblock Can you do a 'keybase log send' so we can see the installer logs? They should show whether our installer caused a reboot.

**Comment by cinderblock** [comment]
@zanderz Log id: 91e86ab5d185db6abb27441c
